chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/SurveillanceReportController.java

```diff
@@ -0,0 +1,220 @@
+        if (!ff4j.check(FeatureList.SURVEILLANCE_REPORTING)) {
+            throw new NotImplementedException();
+        }
+        if (updateRequest.getId() == null) {
+            throw new InvalidArgumentsException(msgUtil.getMessage("report.quarterlySurveillance.missingReportId"));
+        }
+        QuarterlyReportDTO reportToUpdate = reportManager.getQuarterlyReport(updateRequest.getId());
+        //above line throws entity retrieval exception if bad id
+        reportToUpdate.setActivitiesOutcomesSummary(updateRequest.getSurveillanceActivitiesAndOutcomes());
+        reportToUpdate.setPrioritizedElementSummary(updateRequest.getPrioritizedElementSummary());
+        reportToUpdate.setReactiveSummary(updateRequest.getReactiveSummary());
+        reportToUpdate.setTransparencyDisclosureSummary(updateRequest.getTransparencyDisclosureSummary());
+        QuarterlyReportDTO createdReport = reportManager.updateQuarterlyReport(reportToUpdate);
+        return new QuarterlyReport(createdReport);
+    }
+
+    @ApiOperation(value = "Delete a quarterly report.",
+            notes = "Security Restrictions: ROLE_ADMIN or ROLE_ACB and administrative authority "
+            + "on the ACB associated with the report.")
+    @RequestMapping(value = "/quarterly/{quarterlyReportId}", method = RequestMethod.DELETE,
+    produces = "application/json; charset=utf-8")
+    public void deleteQuarterlyReport(@PathVariable final Long quarterlyReportId)
+            throws EntityRetrievalException {
+        if (!ff4j.check(FeatureList.SURVEILLANCE_REPORTING)) {
+            throw new NotImplementedException();
+        }
+        reportManager.deleteQuarterlyReport(quarterlyReportId);
+    }
+
+    @ApiOperation(value = "Download a quarterly report as an XLSX file.",
+            notes = "Security Restrictions: ROLE_ADMIN or ROLE_ACB and administrative authority "
+                    + "on the ACB associated with the report.")
+    @RequestMapping(value = "/export/quarterly/{quarterlyReportId}", method = RequestMethod.GET)
+    public void exportQuarterlyReport(@PathVariable("quarterlyReportId") final Long quarterlyReportId,
+            final HttpServletResponse response) throws EntityRetrievalException, IOException {
+        if (!ff4j.check(FeatureList.SURVEILLANCE_REPORTING)) {
+            throw new NotImplementedException();
+        }
+
+        //create the excel workbook object
+        Workbook reportXlsx = reportManager.exportQuarterlyReport(quarterlyReportId);
+
+        //create a new file in the download directory
+        QuarterlyReportDTO report = reportManager.getQuarterlyReport(quarterlyReportId);
+        String filename = report.getQuarter().getName() + "-" + report.getAnnualReport().getYear()
+                    + "-" + report.getAnnualReport().getAcb().getName() + "-quarterly-report.xlsx";
+        File downloadFile = fileUtils.createDownloadFile(filename);
```

Comment:
Does this mean the download folder will get new files in them permanently on each request for a quarterly surveillance report? That doesn't seem like a good idea. Can you create the Excel file as a temporary file and stream that instead? I remember doing something where we created a temporary file for some other thing in CHPL, though I have no idea where it is